### PR TITLE
ec2: do not try to auth swarm port if exists

### LINKED ISSUE #678: [amazonec2 driver] Attempt to create swarm master bails if SG rule already exists
We get an API error back because the SG rule for swarm already exists, but it's actually fine, it shouldn't affect the attempted creation at all.

screen shot 2015-03-01 at 3 19 04 pm

Edit:  Oops, wrong screenshot.

### PR DESCRIPTION
**Pull request by ehazlett**
Fixes #678

### CODE REVIEW

File: drivers/amazonec2/amazonec2.go [hunk 1]
```diff
@@ -664,42 +658,52 @@ func (d *Driver) configureSecurityGroup(groupName string) error {
 	return nil
 }
 
-func configureSecurityGroupPermissions(group *amz.SecurityGroup) []amz.IpPermission {
+func (d *Driver) configureSecurityGroupPermissions(group *amz.SecurityGroup) []amz.IpPermission {
```
Comment (nathanleclaire) [inline drivers/amazonec2/amazonec2.go]: Ah, the test needs to be updated to reflect that this has been moved to a driver method.

Comment (ehazlett) [inline drivers/amazonec2/amazonec2.go]: ok tests fixed.  thanks!

**Comment by nathanleclaire** [comment]
LGTM

**Comment by ehazlett** [comment]
thanks!

**Pull request merged**